src/ConcordiumNetSdk/Wallets/WalletAccount.cs

```diff
@@ -0,0 +1,139 @@
+///
+/// In particular it supports import of the browser and genesis wallet key export
+/// formats. The class implements <see cref="ITransactionSigner"/> so it may be used
+/// for signing transactions.
+///
+/// This structure does not have the encryption key for sending encrypted transfers, it only
+/// contains keys for signing transactions.
+/// </summary>
+public class WalletAccount : ITransactionSigner
+{
+    /// <summary>
+    /// The address of the imported account.
+    /// </summary>
+    public readonly AccountAddress AccountAddress;
+
+    /// <summary>
+    /// Internal representation of a signer.
+    /// </summary>
+    private TransactionSigner _signer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="WalletAccount"/> class.
+    /// </summary>
+    /// <param name="accountAddress">The address of the imported account.</param>
+    /// <param name="signKeys">The signers corresponding to the keys of the imported account.</param>
+    private WalletAccount(
+        AccountAddress accountAddress,
+        Dictionary<AccountCredentialIndex, Dictionary<AccountKeyIndex, ISigner>> signKeys
+    )
+    {
+        AccountAddress = accountAddress;
+        _signer = new TransactionSigner();
+        signKeys
+            .ToList()
+            .ForEach(cred =>
+            {
+                cred.Value
+                    .ToList()
+                    .ForEach(key =>
+                    {
+                        _signer.AddSignerEntry(cred.Key, key.Key, key.Value);
+                    });
+            });
+    }
+
+    /// <summary>
+    /// Try to create a new instance from a wallet data source.
+    /// </summary>
+    /// <param name="walletDataSource">Data source from which to import the wallet.</param>
+    /// <exception cref="WalletDataSourceException"/>
+    public static WalletAccount From(IWalletDataSource walletDataSource)
+    {
+        return new WalletAccount(
+            walletDataSource.TryGetAccountAddress(),
+            walletDataSource.TryGetSignKeys()
+        );
+    }
+
+    /// <summary>
+    /// Create a new instance from a string in the browser or genesis wallet key export format.
+    /// </summary>
+    /// <param name="json">JSON string in the browser or genesis wallet key export format.</param>
+    /// <exception cref="JsonException">The specified input is not valid JSON.</exception>
+    /// <exception cref="WalletDataSourceException">Either a field is missing or an index or sign key could not be parsed.</exception>
+    public static WalletAccount FromWalletKeyExportFormat(string json)
+    {
+        try
+        {
+            return FromGenesisWalletKeyExportFormat(json);
+        }
+        catch (Exception e)
+        {
+            try
+            {
+                return FromBrowserWalletKeyExportFormat(json);
+            }
+            catch (Exception)
+            {
+                throw e;
+            }
+        }
```

Comment:
I'm usually not a fan of this approach. In particular the browser wallet export explicitly has
```
  "type": "concordium-browser-wallet-account",
```
there to distinguish it from others. So ideally we'd dispatch on that rather than "try and catch" way of doing it.